rest-api/test/unit_test/api_test/participant_summary_api_test.py

```diff
@@ -249,6 +252,89 @@ def testQuery_noSummaries(self):
     response = self.send_get('ParticipantSummary')
     self.assertBundle([], response)
 
+  def test_last_modified_sync(self):
+    SqlTestBase.setup_codes([PMI_SKIP_CODE], code_type=CodeType.ANSWER)
+    questionnaire_id = self.create_demographics_questionnaire()
+    t1 = TIME_1
+    t2 = TIME_1 + datetime.timedelta(seconds=400)
+
+    def setup_participant(when):
```

Comment:
I didn't know you could define functions within functions in Python. :)

We don't do this in other places in RDR code... any particular reason you want to?

Reply:
So FYI, the `def` and the `class` statements are considered valid almost anywhere a statement can be used in Python.  It isn't uncommon to have several layers of internally defined functions in order to take advantage of closures; in fact, that's the key language feature behind how decorators work (unless you go the `__init__` and `__call__` route, using a class as a kind of closed over function).  In fact we use functions inside functions for our decorators; see for example [`auth_required`](https://github.com/all-of-us/raw-data-repository/blob/23c48350f5310800c4c3f1ec60bba70cd28f74b6/rest-api/app_util.py#L173).

In this case I think MM broke the func out just for code organizational purposes; the func closes over `questionnaire_id` but that could easily be a parameter instead and the function could be moved down to the class level.  I'd only bother if we're going to use it in other places though.  Up to ya'll.

Reply:
Fine by me to do things this way. Just noting we hadn't done it (much) before... and that I am still a Python n00b. :)

Reply:
What he said.